oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/endpoint/OAuth2ParameterNames.java

```diff
@@ -60,6 +60,12 @@
 	 */
 	String STATE = "state";
 
+	String CODE_CHALLENGE = "code_challenge";
```

Comment:
I think it might make sense to extend the class and put the extension parameters in a new interface.

Reply:
I suppose that if we subclassed OAuth2AuthorizationRequest with PkceOAuthAuthorizationRequest then it would make sense to separate the PKCE-specific params into PkceParameterNames. However, if we want to back PKCE support directly into OAuth2AuthorizationRequest then OAuth2ParameterNames works? The thing I like about subclassing OAuth2AuthorizationRequest and keeping PkceParameterNames is that it might make the testing of the classes a little cleaner IMO. @rwinch I'll wait for you to chime back in on this. Add PKCE support directly to OAuth2AuthorizationRequest or subclass?